### ISSUE TITLE
Double-click in dom0 on window does not Maximize it (regression from R4.0).

### ISSUE THREAD

**Issue by jamke** [QubesOS/qubes-issues#7880]
### Qubes OS release
R4.1.1.

### Brief summary
Double-click in dom0 on window's title does not Maximize it
It worked for me in R4.0, so it's probably a regression due to domU or other changes.

### Steps to reproduce
1. Set in xfce settings: Windows Manager -> Advanced -> Double click action = Maximize window.
2. Double-click on the window's title.

### Expected behavior
Windows becomes maximized.

### Actual behavior
Nothing happens.

**Comment by unman** [comment]
Works for me.
I didn't need to set anything in Xfce settings.

Also works in KDE

**Comment by jamke** [comment]
@unman I had it work in R4.0. Maybe, you upgraded from it?
I'm talking about clean installation of R4.1.1 with a DomU addon option active during installation (it's default).

Do you have DomU or something? I think it might be a regression reason.

**Comment by jamke** [comment]
> I didn't need to set anything in Xfce settings.

To be clear, I also had this checkbox active in the xfce settings by default, no need to set anything by default.
It is mentioned in the ticket as a point that double click should maximize windows.

**Comment by EncryptedGiraffe** [comment]
Works for me both on XFCE and KDE, by default on clean install

**Comment by tezeb** [comment]
@jamke I had the impression that it does not work too(fresh 4.1 install) and I was used to doing it in 4.0. But after noticing this issue and reports confirming that it in fact works I tried few times and it indeed worked (once in ~10 clicks). It seems that changing double-click speed resolves this for me(System Tools -> Mouse and Touchpad -> Behaviour).

**Comment by andrewdavidwong** [comment]
Also works for me on a clean 4.1.1 installation with Xfce4.

-----

Closing as "cannot reproduce" (we were unable to reproduce this issue). If anyone believes this is a mistake, or if anyone can reproduce the issue, please leave a comment, and we'll be happy to reopen this. Thank you.

**Comment by jamke** [comment]
@tezeb wow, indeed! It works one in several times for me too!

> I had the impression that it does not work too(fresh 4.1 install) and I was used to doing it in 4.0. 
> But after noticing this issue and reports confirming that it in fact works I tried few times and it indeed worked (once in ~10 clicks). 

I set double-click to the same time value (350 ms) as I have on R4.0 on different PC. And still feel that behavior is very different. So, now I have to increase it to 500 to get the same timing as I expect. Not sure why. Maybe because mouse is different but I doubt it.

Thank you for writing your comment, you made feel sane, that I'm not the only one who got this impression that it does not work =)

@andrewdavidwong I agree that issue should be closed.

Maybe the value in Qubes OS should be reconsidered (to 400 ms).

Please, look this ticket (it's 2013):
https://bugzilla.xfce.org/show_bug.cgi?id=10562

It looks that Qubes OS maybe has almost 2 times smaller double-click timing than upstream xfce/kde/gnome (400 ms)? Maybe it sould be changed (I can open different ticket if you think it makes sense).

**Comment by andrewdavidwong** [comment]
This setting should simply be inherited from upstream. I doubt we specifically change it to make it shorter. Maybe Fedora changes the Xfce4 default for some reason?

**Comment by jamke** [comment]
Xfce according to documentation really has **250 ms**, insane.
https://docs.xfce.org/xfce/xfce4-settings/mouse#double_click

This information contradicts this one about it being raised to **400 ms** a long time ago: 
https://bugzilla.xfce.org/show_bug.cgi?id=10562

**Comment by jamke** [comment]
Wow, new information is that - **it is an actually bug, it's not a timing problem**.
People complain that maximizing starts working for them starting from 270 ms only, and never bellow.
https://bugzilla.xfce.org/show_bug.cgi?id=9297

So, @andrewdavidwong @tezeb @EncryptedGiraffe @unman or anybody reading it, please make that test yourself:
1. Go to the `Settings Manager` -> `Mouse and Touchpad` -> `Behavior` and change Double Click time from default 250 ms to 200 ms (not much of a difference really).
2. Click on title or the settings window many times and as fast as possible, way faster than 200 ms and you normally would.
3. What happens?

**In my case no matter how fast and often I double-click with _200 ms_ setting - it _never_ maximizes. Insane!**

I think the issue should be reopened.
Because it's not about me clicking too slow. It's about maximizing not working on clean install or R4.1.1 for me at all.

**Comment by marmarek** [comment]
Interestingly, https://gitlab.xfce.org/xfce/xfwm4/-/issues/92#note_26345
claims it works in xfce 4.14 - the version we have in R4.1.

**Comment by jamke** [comment]
> Interestingly, https://gitlab.xfce.org/xfce/xfwm4/-/issues/92#note_26345
> claims it works in xfce 4.14 - the version we have in R4.1.

I saw it too, but it really does not work for me on R4.1.1 for sure.
My guess is that for some cases or computer it really works, maybe even in older xfce versions, that's why such a problem was not fixed for so long.

P.S. @marmarek have you tried the experiment with **200 ms**, what happened in your case and was maximizing broke for you too?

**Comment by andrewdavidwong** [comment]
It sounds like this has been confirmed to be an upstream bug in Xfce and not a Qubes-specific bug. If that's the case, I'll leave this issue closed as "not our bug" (the Qubes OS Project does not develop the software that is causing this bug). If anyone believes this resolution is a mistake, please leave a comment, and we'll be happy to take another look. Thank you.

Related FAQ: [Why don’t you fix upstream bugs that affect Qubes OS?](https://www.qubes-os.org/faq/#why-dont-you-fix-upstream-bugs-that-affect-qubes-os)

**Comment by jamke** [comment]
Just for documentation purposes: 
Yesterday I had this issue with maximizing completely not working **even with a big value of double-click time (400-500 ms)**.
The window title was completely ignoring double-clicks for unknown reason.

**Comment by jamke** [comment]
Once again, on the latest completely up-to-date Qubes OS R4.1.1 - double clicking is not working at all, with any double-click time.
P.S. Maybe XFCE should be replaced with KDE, it's a disaster.

**Comment by jamke** [comment]
I have this problem on R4.3.4. 
Please reopen.

**Comment by jamke** [comment]
Technically the issue is not only with dom0 windows, but windows of all qubes. Maximize breaks in R4.2.4 in xfce as it was before in xfce of R4.1, R4.0.
On the other hand all windows in dom0 are managed by dom0, so I probably should keep the title unchanged.

**Comment by jamke** [comment]
@andrewdavidwong well, I supposedly have xfce 4.16+ in R4.2.4, and the upstream issue is excepted to be fixed in xfce 4.14. 
BTW, I see no version numbers of xfce in `About Xfce` dialog, surprising. 

Maybe (probably) it is still upstream, well, because it's xfce, but I am not sure.

**Comment by marmarek** [comment]
> BTW, I see no version numbers of xfce in `About Xfce` dialog, surprising.

There is "Xfce Version" field in the "System" tab (the first one) in that dialog. And it says 4.16.

I just checked, and double click for maximize works for me with default 250ms double click time. When I changed it to 200ms, then it still works fine on some windows (like the mouse settings window), but on some others (like xfce4-terminal) it's hard to get (but still manage to get it maximized after 2 or 3 tries).
Anyway, still feels like upstream issue, and pretty minor IMHO (given default config works).

**Comment by jamke** [comment]
> There is "Xfce Version" field in the "System" tab (the first one) in that dialog. And it says 4.16.

True, I did not notice it between hardware specs. I have 4.16, too.

The double-click to maximize is not working for me even with 500 ms setting.
